### ISSUE TITLE
Yellow and white flashings w/ D435

### ISSUE THREAD

**Issue by tangbao** [vimeo/vimeo-depth-viewer#6]
Hi,

I use your pre-compiled viewer and the viewer compiled by myself to test, but none of them works. Here is a snapshot. The left window is always with yellow and white flashings and the right window sometimes shows the same flashings, or sometimes shows videos with many noises.

![image](https://user-images.githubusercontent.com/12832546/64745386-b0275100-d4d5-11e9-8acb-c464cc5c110a.png)

 I change do not change the code, and I have no idea what is wrong. Could you please give me some suggestions?

I am using:

- Macbook Air 2015 (Majove 10.14.6)
- Cmake 3.15.3
- Intel Realsense D435

Thank you very much

**Comment by juniorxsound** [comment]
I think this could be related to the USB bus on your MBA 2015. It is not USB 3.0 right? I also wonder if that happens when using the [Intel Realsense Viewer](https://github.com/IntelRealSense/librealsense/tree/master/tools/realsense-viewer) that comes in the SDK

**Comment by tangbao** [comment]
> I think this could be related to the USB bus on your MBA 2015. It is not USB 3.0 right? I also wonder if that happens when using the [Intel Realsense Viewer](https://github.com/IntelRealSense/librealsense/tree/master/tools/realsense-viewer) that comes in the SDK

Thank you for your quick reply. Sorry that I forgot to mention the camera works well with the Intel Realsense Viewer. And I think the USB bus on MBA 2015 is USB 3.0. Thanks!

![image](https://user-images.githubusercontent.com/12832546/64746295-0053e280-d4d9-11e9-8e5c-658ac6696077.png)

**Comment by tangbao** [comment]
Could you please provide the firmware version of your D415 and D435? Maybe the old SDK you use does not support the newest firmware. Thanks!

**Comment by juniorxsound** [comment]
Hi @tangbao was able to build and run this on MBP 2018 (Mojave 10.14.6) and run successfully with a D415
<img width="400" alt="Screen Shot 2019-09-13 at 8 20 53 PM" src="https://user-images.githubusercontent.com/15909884/64901154-51441200-d664-11e9-9ae9-c4878c529066.png">
One quick caveat I did catch is it failed CMak'ing the project due to a `librealsense` GLFW issue so had to turn off the visual examples when using CMake - `cmake ../ -DBUILD_GRAPHICAL_EXAMPLES=Off`

**Comment by tangbao** [comment]
Thanks @juniorxsound ! Have you tried it with D435?

**Comment by juniorxsound** [comment]
Unfortunately I do not have one, so can't really test it but if you want to dive into the code and try to debug that the `librealsense` pipeline initialization [happens here](https://github.com/vimeo/vimeo-depth-viewer/blob/master/src/MainWindow.cpp#L284)

**Comment by tangbao** [comment]
Hi @juniorxsound ,

Thank you for your suggestions. I solved the GLFW issue by installing GLFW on my machine before. I received the D415 camera today but it still not worked.... I tried to debug it but I failed to find anything wrong. Maybe I need to use a newer machine.

Thanks!

**Comment by tangbao** [comment]
Hi @juniorxsound 

I tried on an MBP 2017 today and it worked perfectly both with D415 and D435. 

Now I am playing with your depth-player now. It seems that it needs some tricks (like adjusting the angle of camera and the size of window in OBS) to let the player get a good rendering result. Also, the streaming thing does not work now because the XMLHttpRequest is banned by CORS policy. I will look into this problem tomorrow.

One more thing -- the 3D hologram is generated based on 3D mesh, not point cloud, right?

Thank you for your time so much.

**Comment by juniorxsound** [comment]
@tangbao Correct, it starts with a fully tesselated grid and displaces it using the depth map
